thrust/CMakeLists.txt

```diff
@@ -59,6 +59,16 @@ option(THRUST_ENABLE_TESTING "Build Thrust testing suite." "ON")
 option(THRUST_ENABLE_EXAMPLES "Build Thrust examples." "ON")
 option(THRUST_ENABLE_BENCHMARKS "Build Thrust runtime benchmarks." "${CCCL_ENABLE_BENCHMARKS}")
 
+# Force the offset type to 32bit. Improves compile times and binary size at the expense of limiting input sizes
+option(THRUST_FORCE_32BIT_OFFSET_TYPE "Requires thrust to use 32 bit offset types." "OFF")
+
+# Force the offset type to 64bit. Improves compile times and binary size potentially degrading runtime performance
+option(THRUST_FORCE_64BIT_OFFSET_TYPE "Requires thrust to use 64 bit offset types." "OFF")
+
+if (THRUST_FORCE_32BIT_OFFSET_TYPE AND THRUST_FORCE_64BIT_OFFSET_TYPE)
+  message(FATAL_ERROR "Only THRUST_FORCE_32BIT_OFFSET_TYPE or THRUST_FORCE_64BIT_OFFSET_TYPE may be defined!")
+endif()
```

Comment:
Suggestion: I think I advocated in the past, but forgot where, to have a tristate option here: `THRUST_OFFSET_TYPE`  with values `AUTO` (default), `32` and `64`. I don't mind the names.

Reply:
`option` is a boolean so we would have to do something like
```cmake
    set(THRUST_DISPATCH_TYPE "Default" CACHE STRING "Choose the type of thrust dispatch." FORCE)
    # Set the possible values for ccmake and cmake-gui
    set_property(CACHE THRUST_DISPATCH_TYPE PROPERTY STRINGS "Default" "Force32bit" "Force64bit")
```
And then later
```cmake
  if (THRUST_DISPATCH_TYPE == "Force32bit")
    list(APPEND cxx_compile_definitions "THRUST_FORCE_32BIT_OFFSET_TYPE")
  elseif (THRUST_DISPATCH_TYPE == "Force64bit")
    list(APPEND cxx_compile_definitions "THRUST_FORCE_64BIT_OFFSET_TYPE")
  endif(
```

I am undecided on what is better

Reply:
I am undecided as well, since for the C++ land testing for the precense of a macro like `THRUST_FORCE_32BIT_OFFSET_TYPE` is more common than checking `THRUST_OFFSET_TYPE == THRUST_AUTO`. Also, the latter implies that `THRUST_OFFSET_TYPE` is always defined, unless we say that the absence of `THRUST_OFFSET_TYPE` means automatic choice.

I also considered a macro `THRUST_OVERRIDE_OFFSET_TYPE` which is either undefined (automatic choice) or defines what type to use.

In any case, I have no strong preference for any of these options.

Reply:
Let's keep the current implementation. This isn't going to be a common user facing thing and the current approach is satisfactory.